### ISSUE TITLE
[Bug]: StrictMode spams logs with WindowManager Errors

### ISSUE THREAD

**Issue by vfutia** [Appboy/appboy-android-sdk#222]
### Braze Android SDK Version

15.0.0

### Steps To Reproduce

Use an app integrated with the Braze SDK on an Android device with Android 11+

### Expected Behavior

Logs are not spammed

### Actual Incorrect Behavior

The logs are spammed

### Verbose Logs

```shell
2022-04-27 11:51:01.088 1654-1840/ E/ContextImpl: Tried to access visual service WindowManager from a non-visual Context:com.udemy.android.CombinedApplication@ddf480 Visual services, such as WindowManager, WallpaperService or LayoutInflater should be accessed from Activity or other visual Context. Use an Activity or a Context created with Context#createWindowContext(int, Bundle), which are adjusted to the configuration and visual bounds of an area on screen.
    java.lang.IllegalAccessException: Tried to access visual service WindowManager from a non-visual Context:com.udemy.android.CombinedApplication@ddf480
        at android.app.ContextImpl.getSystemService(ContextImpl.java:1914)
        at android.content.ContextWrapper.getSystemService(ContextWrapper.java:803)
        at bo.app.n1.a(SourceFile:25)
        at bo.app.n1.e(SourceFile:7)
        at bo.app.n1.b(SourceFile:1)
        at bo.app.u.a(SourceFile:35)
        at bo.app.u.b(SourceFile:17)
        at bo.app.u.e(SourceFile:2)
        at bo.app.r$b.run(SourceFile:4)
        at java.lang.Thread.run(Thread.java:923)
```

### Additional Information

I have tried upgrading to the latest version of the Braze SDK in hopes that this wouldn't happen anymore but didn't have any luck.  This isn't an _error_ per se, but having the logs filled up with all of this is really annoying.  This gets put into the logs repeatedly throughout usage of the app.

**Comment by radixdev** [comment]
Hi @vfutia ,

Can you provide the StrictMode configuration used to produce this issue?

**Comment by vfutia** [comment]
I would be happy to!

```
handler.postAtFrontOfQueue(new Runnable() {
            @Override
            public void run() {
                //to avoid strict mode problem
                StrictMode.setThreadPolicy(new StrictMode.ThreadPolicy.Builder()
                        .detectAll()
                        .penaltyLog()
                        .build());
            }
        });
        handler.postAtFrontOfQueue(new Runnable() {
            @Override
            public void run() {
                StrictMode.VmPolicy.Builder builder = new StrictMode.VmPolicy.Builder();
                builder.detectAll();
                if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
                    final ExecutorService EXECUTOR = Executors.newSingleThreadExecutor();
                    builder.penaltyListener(EXECUTOR, v -> {
                        // disabling "IncorrectContextUseViolation", to be removed when we start supporting multi-display devices
                        if ((v instanceof UntaggedSocketViolation) || v.getStackTrace()[0].getMethodName().equals("onIncorrectContextUsed")) {
                            // do nothing
                        } else {
                            Timber.e(v);
                        }
                    });
                } else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
                    final ExecutorService EXECUTOR = Executors.newSingleThreadExecutor();
                    builder.penaltyListener(EXECUTOR, v -> {
                        // exclude UntaggedSocketViolation
                        if (!(v instanceof UntaggedSocketViolation)) {
                            Timber.e(v);
                        }
                    });
                } else {
                    builder.penaltyLog();
                }
                StrictMode.setVmPolicy(builder.build());
            }
        });
    }
```

**Comment by radixdev** [comment]
Hi @vfutia ,

Our use of the application context throughout the SDK (versus a lifecycle dependent context such as Activity) was a deliberate choice in order to not run into Activity leak issues (which is another StrictMode violation). In addition, rearchitecting our use of context could result in situations where the "live" context is unavailable and result in suboptimal analytics or other display issues to users. 

While I know this is not ideal, I would recommend disabling `onIncorrectContextUsed()` in the policy builder at this time. I would also note that we can reevaluate the situation if the Activity leakage issue can be alleviated or if the Android OS upgrades this warning to a runtime error in a future release.

**Comment by vfutia** [comment]
We're already filtering that out here

```
if ((v instanceof UntaggedSocketViolation) || v.getStackTrace()[0].getMethodName().equals("onIncorrectContextUsed")) {
       // do nothing
}
```

**Comment by allco** [comment]
I have same issue on Android 12 + Braze 15.0.0

```
431-3574 E/ContextImpl: Tried to access visual service WindowManager from a non-visual Context:net.easypark.android.DebugMobileApp@7af916e Visual services, such as WindowManager, WallpaperService or LayoutInflater should be accessed from Activity or other visual Context. Use an Activity or a Context created with Context#createWindowContext(int, Bundle), which are adjusted to the configuration and visual bounds of an area on screen.
    java.lang.IllegalAccessException: Tried to access visual service WindowManager from a non-visual Context:net.easypark.android.DebugMobileApp@7af916e
        at android.app.ContextImpl.getSystemService(ContextImpl.java:1916)
        at android.content.ContextWrapper.getSystemService(ContextWrapper.java:803)
        at bo.app.n1.a(SourceFile:25)
        at bo.app.n1.e(SourceFile:7)
        at bo.app.n1.b(SourceFile:1)
        at bo.app.u.a(SourceFile:35)
        at bo.app.u.b(SourceFile:17)
        at bo.app.u.e(SourceFile:2)
        at bo.app.r$b.run(SourceFile:4)
        at java.lang.Thread.run(Thread.java:923)
 ```

**Comment by radixdev** [comment]
Hi @vfutia @allco @karthik-cherryz @ursimon, ad @nick-titov-sw,

We've identified a fix and will include it in the next release. We'll update this issue once that happens 🎉

**Comment by radixdev** [comment]
Hi all,

We've introduced a fix for this in the latest SDK, 24.3.0.
